src/charm.py

```diff
@@ -347,6 +370,41 @@ def external_url(self) -> str:
     # === UTILITY METHODS === #
     ###########################
 
+    def _render_workers_alert_rules(self):
+        self._remove_rendered_rules()
+
+        apps = set()
+        for worker in self.cluster_provider.gather_topology():
+            if worker["app"] in apps:
+                continue
+
+            apps.add(worker["app"])
+            topology_dict = {
+                "model": self.model.name,
+                "model_uuid": self.model.uuid,
+                "application": worker["app"],
+                "unit": worker["unit"],
+                "charm_name": "mimir-worker-k8s",
+            }
+            topology = JujuTopology.from_dict(topology_dict)
+            alert_rules = AlertRules(query_type="promql", topology=topology)
+            alert_rules.add_path(WORKER_ORIGINAL_ALERT_RULES_PATH, recursive=True)
+            alert_rules_contents = yaml.dump(alert_rules.as_dict())
+
+            file_name = f"{CONSOLIDATED_ALERT_RULES_PATH}/rendered_{worker['app']}.rules"
+            with open(file_name, "w") as writer:
+                writer.write(alert_rules_contents)
+
+    def _remove_rendered_rules(self):
+        files = glob.glob(f"{CONSOLIDATED_ALERT_RULES_PATH}/rendered_*")
+        for f in files:
+            os.remove(f)
```

Comment:
Should we use [rmtree](https://docs.python.org/3/library/shutil.html#shutil.rmtree)?

Reply:
rmtree deletes an entire directory tree, here we are only deleting files starting with "rendered_".